pkg/cliplugins/kubestellar/get_log/cmd/cmd.go

```diff
@@ -0,0 +1,219 @@
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+	"k8s.io/client-go/util/homedir"
+	"k8s.io/klog/v2"
+)
+
+var (
+	kubeconfig string
+	followFlag bool
+	linesFlag  string
+	whichArg   string
+	logger     klog.Logger
+)
+
+func New(streams genericclioptions.IOStreams) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get-log [flags] which",
+		Short: "Get the log output from a central KubeStellar or kcp process",
+		Args:  cobra.ExactArgs(1),
+		Run:   getLog,
+	}
+
+	// Create a new pflag.FlagSet for custom flags
+	fs := pflag.NewFlagSet("get-log", pflag.ExitOnError)
+	fs.StringVarP(&kubeconfig, "kubeconfig", "c", "", "Path to the kubeconfig file")
+	fs.BoolVarP(&followFlag, "follow", "f", false, "Follow the log output")
+	fs.StringVar(&linesFlag, "lines", "+0", "Number of lines to display")
+	fs.Parse(os.Args[1:])
+
+	// Add custom pflag.FlagSet to the Cobra command
+	cmd.Flags().AddFlagSet(fs)
+
+	// Create a genericclioptions.ConfigFlags instance
+	cliOpts := genericclioptions.NewConfigFlags(true)
+	cliOpts.AddFlags(cmd.PersistentFlags())
+
+	ctx := context.Background()
+	logger := klog.FromContext(ctx)
+	ctx = klog.NewContext(ctx, logger)
+	fs.VisitAll(func(flg *pflag.Flag) {
+		logger.V(1).Info("Command line flag", flg.Name, flg.Value)
+	})
+
+	return cmd
+}
+
+func getLog(cmd *cobra.Command, args []string) {
+	whichArg = args[0]
+	ksNamespace := "kubestellar"
+	ksSelector := "app=kubestellar-server"
+
+	logfile := ""
+	switch whichArg {
+	case "kcp":
+		logfile = "kcp.log"
+	case "where-resolver":
+		logfile = "kubestellar-where-resolver-log.txt"
+	case "mailbox-controller":
+		logfile = "mailbox-controller-log.txt"
+	case "placement-translator":
+		logfile = "placement-translator-log.txt"
+	default:
+		fmt.Println("Argument 1 must be one of: kcp, where-resolver, mailbox-controller, placement-translator")
+		os.Exit(1)
+	}
+
+	// Build the Kubernetes clientset
+	// config, err := cliOpts.ToRESTConfig()
+	// if err != nil {
+	// 	logger.Error(err, "Failed to build config from flags")
+	// 	os.Exit(5)
+	// }
+
+	config, err := buildConfig()
+	if err != nil {
+		fmt.Println("Error building Kubernetes config:", err)
+		os.Exit(1)
+	}
+	clientset, err := kubernetes.NewForConfig(config)
+	if err != nil {
+		klog.Error("Error creating Kubernetes clientset:", err)
+		os.Exit(10)
+	}
+	namespace, err := getNamespace(clientset, ksNamespace)
+	if err != nil {
+		klog.Error("Error getting server namespace:", err)
+		os.Exit(15)
+	}
+	podName, err := getServerPodName(clientset, namespace, ksSelector)
+	if err != nil {
+		klog.Error("Error getting server pod name:", err)
+		os.Exit(20)
+	} else {
+		podName = strings.Trim(string(podName), "'")
```

Comment:
Do you have evidence that this Trim is necessary?  My understanding of the API used in `getServerPodName` is that it returns names without quotes around them.